app/lib/features/chat_ng/widgets/reactions/reaction_detail_sheet.dart

```diff
@@ -0,0 +1,221 @@
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/common/themes/app_theme.dart';
+import 'package:acter/features/chat_ng/providers/chat_room_messages_provider.dart';
+import 'package:acter_avatar/acter_avatar.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+class ReactionDetailsSheet extends ConsumerStatefulWidget {
+  final String roomId;
+  final List<ReactionItem> reactions;
+
+  const ReactionDetailsSheet({
+    super.key,
+    required this.roomId,
+    required this.reactions,
+  });
+
+  @override
+  ConsumerState<ReactionDetailsSheet> createState() =>
+      _ReactionDetailsSheetState();
+}
+
+class _ReactionDetailsSheetState extends ConsumerState<ReactionDetailsSheet>
+    with TickerProviderStateMixin {
+  late TabController _tabController;
+  late List<Tab> _tabs;
+
+  @override
+  void initState() {
+    super.initState();
+    _initializeTabs();
+  }
```

Comment:
I understand that the tabs controller needs a stateful widget and the crap, _but_ then we also need to rebuild that list upon didWidgetChange, no? Otherwise we have new emojis coming in and show them in the render, but don't have the reactions in the tabs ...?!?